### PR TITLE
Move ScriptStore and RuntimeHolder from RN to RNW.

### PR DESCRIPTION
**Pull request by hansenyy**
This PR moves ScriptStore and RuntimeHolder from RN to RNW. See also https://github.com/microsoft/react-native/pull/185.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3577)

### CODE REVIEW

File: vnext/JSI/Shared/RuntimeHolder.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <jsi/jsi.h>
+#include <memory>
+
+namespace facebook {
```
Comment (NikoAri) [inline vnext/JSI/Shared/RuntimeHolder.h]: >{ [](start = 19, length = 1)

(nit) why not use latest faceboo::jsi syntax for namespaces?

File: vnext/JSI/Shared/RuntimeHolder.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+namespace jsi {
+
+// An instance of this interface is expected to
```
Comment (NikoAri) [inline vnext/JSI/Shared/RuntimeHolder.h]: >// [](start = 0, length = 2)

(nit) We should have proper documetation standard at least for new code, eg, doxygen. So comments would be:
//!

File: vnext/JSI/Shared/RuntimeHolder.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// a. lazily create a JSI Runtime on the first call to getRuntime
+// b. subsequent calls to getRuntime should return the Runtime created in (a)
+
+// Note :: All calls to getRuntime() should happen on the same thread unless you are sure that
```
Comment (NikoAri) [inline vnext/JSI/Shared/RuntimeHolder.h]: >the same thread [](start = 54, length = 15)

(nit) do we care to have code that asserts this?

File: vnext/JSI/Shared/ScriptStore.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      const ScriptSignature &scriptMetadata,
+      const JSRuntimeSignature &runtimeMetadata,
+      const char *prepareTag // Optional tag. For e.g. eagerly evaluated vs lazy cache.
+      ) noexcept = 0;
+};
+
+// JSI::Runtime implementation must be provided an instance on this interface to enable version sensitive capabilities
+// such as usage of pre-prepared javascript script. Alternatively, this entity can be used to directly provide the
+// Javascript buffer and rich metadata to the JSI::Runtime instance.
+struct ScriptStore {
```
Comment (NikoAri) [inline vnext/JSI/Shared/ScriptStore.h]: >ScriptStore [](start = 7, length = 11)

Aren't we planning to get rid of this per discussion last week?

**Comment by NikoAri** [comment]
> #include <string>

(nit) can we alphabetize these?

---
Refers to: vnext/ReactUWP/Utils/UwpPreparedScriptStore.h:7 in 03f1b03. [](commit_id = 03f1b03e589298841225654b4048feee4ff559ad, deletion_comment = False)

File: vnext/ReactWindowsCore/ChakraRuntimeHolder.h [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
 #include <DevSettings.h>
 
-#include <jsi/RuntimeHolder.h>
-#include <jsi/ScriptStore.h>
+#include <JSI/Shared/RuntimeHolder.h>
+#include <jsi/Shared/ScriptStore.h>
```
Comment (NikoAri) [inline vnext/ReactWindowsCore/ChakraRuntimeHolder.h]: >jsi/ [](start = 10, length = 4)

(nit) please be consistent in casing

File: vnext/JSI/Shared/ScriptStore.h (hunk 4 above)
Comment (hansenyy) [reply vnext/JSI/Shared/ScriptStore.h]: Yes. This PR is just moving RuntimeHolde.h and ScriptStore.h from RN to RNW and I am planning on making more changes here. I'll also fix the nit comments then.

---
In reply to: [341760394](https://github.com/microsoft/react-native-windows/pull/3577#discussion_r341760394) [](ancestors = 341760394)

**Comment by hansenyy** [comment]
> #include <string>

This is strange. I thought yarn format should have already done this?

---
In reply to: [548959995](https://github.com/microsoft/react-native-windows/pull/3577#issuecomment-548959995) [](ancestors = 548959995)

---
Refers to: vnext/ReactUWP/Utils/UwpPreparedScriptStore.h:7 in 03f1b03. [](commit_id = 03f1b03e589298841225654b4048feee4ff559ad, deletion_comment = False)

**Pull request merged**